src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/PaperResponseAdmissionCallbackHandlerTest.java

```diff
@@ -0,0 +1,366 @@
+    private CaseMapper caseMapper;
+    @Mock
+    private CaseDetailsConverter caseDetailsConverter;
+    @Mock
+    private DocAssemblyService docAssemblyService;
+    @Mock
+    private DocAssemblyResponse docAssemblyResponse;
+    @Mock
+    private DocumentManagementService documentManagementService;
+    @Mock
+    private Clock clock;
+    @Mock
+    private UserService userService;
+    @Mock
+    private DocAssemblyTemplateBodyMapper docAssemblyTemplateBodyMapper;
+    private UserDetails userDetails;
+
+    @BeforeEach
+    public void setUp() {
+        String paperResponseAdmissionTemplateId = "CV-CMC-GOR-ENG-0016.docx";
+        handler = new PaperResponseAdmissionCallbackHandler(caseDetailsConverter,
+            defendantResponseNotificationService, caseMapper, docAssemblyService, docAssemblyTemplateBodyMapper,
+            paperResponseAdmissionTemplateId, userService, documentManagementService, clock);
+        CallbackRequest callbackRequest = CallbackRequest
+            .builder()
+            .caseDetails(CaseDetails.builder().data(Collections.EMPTY_MAP).build())
+            .eventId(CaseEvent.PAPER_RESPONSE_ADMISSION.getValue())
+            .build();
+        callbackParams = CallbackParams.builder()
+            .type(CallbackType.ABOUT_TO_SUBMIT)
+            .request(callbackRequest)
+            .params(ImmutableMap.of(CallbackParams.Params.BEARER_TOKEN, AUTHORISATION))
+            .build();
+
+        userDetails = SampleUserDetails.builder()
+            .withForename("Forename")
+            .withSurname("Surname")
+            .withRoles("caseworker-cmc")
+            .build();
+    }
+
+    @Test
+    void shouldChangeFileNameForOCON9xFormForFullAdmission() {
+
+        CCDCase ccdCase = getCCDCase(FULL_ADMISSION, CCDRespondent.builder(), "OCON9x");
+
+        Claim claim = Claim.builder()
+            .referenceNumber("XXXXX")
+            .build();
+        when(caseMapper.from(any(CCDCase.class))).thenReturn(claim);
+        when(userService.getUserDetails(AUTHORISATION)).thenReturn(userDetails);
+
+        when(docAssemblyService
+            .renderTemplate(any(CCDCase.class), anyString(), anyString(), any(DocAssemblyTemplateBody.class)))
+            .thenReturn(docAssemblyResponse);
+        when(docAssemblyTemplateBodyMapper.paperResponseAdmissionLetter(any(CCDCase.class), any(String.class)))
+            .thenReturn(DocAssemblyTemplateBody.builder().build());
+        when(docAssemblyResponse.getRenditionOutputLocation()).thenReturn(DOC_URL);
+        when(documentManagementService.getDocumentMetaData(anyString(), anyString())).thenReturn(getLinks());
+        when(clock.instant()).thenReturn(LocalDate.parse("2020-06-22").atStartOfDay().toInstant(ZoneOffset.UTC));
+        when(clock.getZone()).thenReturn(ZoneOffset.UTC);
+        when(clock.withZone(LocalDateTimeFactory.UTC_ZONE)).thenReturn(clock);
+        when(caseDetailsConverter.extractCCDCase(any(CaseDetails.class))).thenReturn(ccdCase);
```

Comment:
Looks like these are repeated in this and `shouldChangeFileNameForOCON9xFormForPartAdmission`. You should move this into a nested sub class and put this in a @BeforeEach method. Give me a shout if you want help.